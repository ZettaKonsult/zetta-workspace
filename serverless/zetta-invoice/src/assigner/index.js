/* @flow */

/**
 * @date 2018-01-16
 */
import getAssignments from './getAssignments';

export { getAssignments };
