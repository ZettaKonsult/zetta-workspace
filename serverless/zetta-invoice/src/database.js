/* @flow */

import AWS from 'aws-sdk';

const isOffline = process.env.IS_OFFLINE;
console.log(isOffline);

const dynamodbOfflineOptions = {
  region: 'localhost',
  endpoint: 'http://localhost:8000',
};

const client = isOffline
  ? new AWS.DynamoDB.DocumentClient(dynamodbOfflineOptions)
  : new AWS.DynamoDB.DocumentClient();

export const getDbTable = (params: { name: string }): string => {
  return params.name;
};

export default async (method: string, params: { [string]: any }) => {
  return await client[method](params).promise();
};
