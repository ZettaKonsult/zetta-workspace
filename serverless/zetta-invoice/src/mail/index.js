/* @flow */

/**
 * @date 2018-03-12
 */

import invoicePDF from './mail';
import { send, prepareAndSend, sendInvoice } from './emailDoc';
export default { invoicePDF, send, sendInvoice, prepareAndSend };
