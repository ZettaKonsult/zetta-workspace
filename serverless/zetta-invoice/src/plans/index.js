/* @flow */

/**
 * @date 2018-03
 */

import get from './get';
import list from './list';
import save from './save';
import remove from './remove';
import { updateRecipientIds } from './update';

export default { get, list, save, updateRecipientIds, remove };
