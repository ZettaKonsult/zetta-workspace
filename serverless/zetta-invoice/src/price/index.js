/**
 * @date 2018-04-10
 */

import calculate from './price';

export default { calculate };
