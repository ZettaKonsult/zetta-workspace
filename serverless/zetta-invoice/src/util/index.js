import parser from './parser';
import { getS3Object, putS3Object } from './s3';
import db, { getDbTable } from './database';
import { failure, success } from './response';
import request from './http';

export {
  parser,
  request,
  getS3Object,
  putS3Object,
  db,
  getDbTable,
  failure,
  success,
};
