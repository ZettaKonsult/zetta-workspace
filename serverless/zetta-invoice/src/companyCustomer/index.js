/* @flow */

import companyCustomerDatabase from './companyCustomerDatabase';
import { db, getDbTable } from '../util';

const companyCustomerTable = getDbTable({ name: 'CompanyCustomers' });

export default companyCustomerDatabase(db)(companyCustomerTable);
