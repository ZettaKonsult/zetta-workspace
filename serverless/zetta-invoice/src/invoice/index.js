/* @flow */

/**
 * @date 2017-02-13
 */
import invoice from './invoice';
import invoiceDatabase from './invoiceDynamodb';
import invoiceGroupDatabase from './invoiceGroupDatabase';

import { db } from '../util';
import { getDbTable } from '../util/database';

const TableNameInvoice = getDbTable({ name: 'Invoices' });
const databaseInvoice = invoiceDatabase(db)(TableNameInvoice);

const TableNameInvoiceGroup = getDbTable({ name: 'InvoiceGroups' });
const databaseInvoiceGroup = invoiceGroupDatabase(db)(TableNameInvoiceGroup);

export default {
  create: data => invoice().create(data),
  invoiceDatabase: databaseInvoice,
  databaseInvoiceGroup,
};
