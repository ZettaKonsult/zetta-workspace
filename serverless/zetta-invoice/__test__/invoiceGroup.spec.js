/**
 * @date 2018-03-05
 */
import { request, testConfig } from '../src/util/';

const host = testConfig.Host;

describe('InvoiceGroup', () => {
  let invoiceGroup = {};

  it('create a group', async () => {
    const data = {
      recipient: {
        name: 'testGroup',
        value: 100,
      },
    };

    invoiceGroup = await request({
      host,
      path: 'invoice/group',
      payload: {
        method: 'post',
        body: data,
      },
    });

    expect({ ...invoiceGroup }).toMatchSnapshot({
      id: expect.any(String),
    });
  });

  it('delete group', async () => {
    const data = { id: invoiceGroup.id };

    const result = await request({
      host,
      path: 'invoice/group',
      payload: {
        method: 'delete',
        body: data,
      },
    });

    expect(result).toBeTruthy();
  });
});
