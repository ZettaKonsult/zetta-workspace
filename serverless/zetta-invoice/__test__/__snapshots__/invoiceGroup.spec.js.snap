// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceGroup create a group 1`] = `
Object {
  "companyCustomerId": "offlineContext_cognitoIdentityId",
  "currentId": 0,
  "id": Any<String>,
  "name": "default",
}
`;
